sklearn/metrics/cluster/_supervised.py

```diff
@@ -138,6 +146,147 @@ def contingency_matrix(labels_true, labels_pred, *, eps=None, sparse=False):
 
 # clustering measures
 
+def pair_confusion_matrix(labels_true, labels_pred):
+    """Pair confusion matrix arising from two clusterings.
+
+    The pair confusion matrix computes a 2x2 similarity matrix between
+    two clusterings by considering all pairs of samples and counting
+    pairs that are assigned in the same or different clusters in the
+    predicted and true clusterings.
+
+    Read more in the :ref:`User Guide <pair_confusion_matrix>`.
+
+    Parameters
+    ----------
+    labels_true : int array, shape = [n_samples]
+        Ground truth class labels to be used as a reference
+
+    labels_pred : array, shape = [n_samples]
+        Cluster labels to evaluate
+
+    Returns
+    -------
+    array, shape(2, 2)
```

Comment:
we need to be clear on the semantics of each axis. They should correspond to confusion matrix: axis 0 should represent the true clustering, and axis 1 the predicted clustering

Reply:
- I used pair_confusion_matrix and your variables in adjusted_rand_index and made it part of the PR, and added that Steinley2004 reference both to the doc string and to the user guide.
- I updated the doc string and user guide for pair_confusion_matrix following the doc of confusion_matrix making the semantics clear.
- I went back to the original (and faster) implementation of pair_confusion_matrix without _comb2 as pair_confusion_matrix now was the only method in _supervised.py using _comb2, and I removed the comb / _comb2 methods altogether.